src/main/java/com/sngular/kloadgen/extractor/impl/SchemaExtractorImpl.java

```diff
@@ -60,47 +55,48 @@ public SchemaExtractorImpl(final AvroExtractor avroExtractor, final JsonExtracto
+        case JSON:
+          response = jsonExtractor;
+          break;
+        case AVRO:
+          response = avroExtractor;
+          break;
+        case PROTOBUF:
+          response = protoBufExtractor;
+          break;
+        default:
+          throw new KLoadGenException(String.format("Schema type not supported %s", schemaType));
+      }
+      return response;
+    } else {
+      throw new KLoadGenException(String.format("Schema type not supported %s", schemaType));
+    }
+  }
+
   @Override
   public final Pair<String, List<FieldValueMapping>> flatPropertiesList(final String subjectName) throws IOException, RestClientException {
     String schemaType = null;
-    Properties properties = JMeterContextService.getContext().getProperties();
+    final Properties properties = JMeterContextService.getContext().getProperties();
 
     final var schemaParsed = JMeterHelper.getParsedSchema(subjectName, properties);
-    final List<FieldValueMapping> attributeList = new ArrayList<>();
+    List<FieldValueMapping> attributeList = new ArrayList<>();
 
-    String registryName = properties.getProperty(SchemaRegistryKeyHelper.SCHEMA_REGISTRY_NAME);
+    final String registryName = properties.getProperty(SchemaRegistryKeyHelper.SCHEMA_REGISTRY_NAME);
     if (Objects.nonNull(registryName) && registryName.equalsIgnoreCase(SchemaRegistryEnum.APICURIO.toString())) {
-      ApicurioParsedSchema apicurioParsedSchema = (ApicurioParsedSchema) schemaParsed;
-      Object schema = apicurioParsedSchema.getSchema();
+      final ApicurioParsedSchema apicurioParsedSchema = (ApicurioParsedSchema) schemaParsed;
+      final Object schema = apicurioParsedSchema.getSchema();
 
       schemaType = apicurioParsedSchema.getType();
-      if (SchemaTypeEnum.AVRO.name().equalsIgnoreCase(schemaType)) {
-        ((Schema) schema).getFields().forEach(field -> avroExtractor.processField(field, attributeList, true, false));
-      } else if (SchemaTypeEnum.JSON.name().equalsIgnoreCase(schemaType)) {
-        attributeList.addAll(jsonExtractor.processSchema(((io.apicurio.registry.serde.jsonschema.JsonSchema) schema).toJsonNode()));
-      } else if (SchemaTypeEnum.PROTOBUF.name().equalsIgnoreCase(schemaType)) {
-        final var protoFileElement = ((io.apicurio.registry.utils.protobuf.schema.ProtobufSchema) schema).getProtoFileElement();
-        // todo: en vez de null comprobar si hay que crear un HashMap con field.getNestedTypes()
-        protoFileElement.getTypes().forEach(field -> protoBufExtractor.processField(field, attributeList, protoFileElement.getImports(), false, null));
-      } else {
-        throw new KLoadGenException(String.format("Schema type not supported %s", apicurioParsedSchema.getType()));
-      }
-
+      attributeList = getExtractor(schemaType).processApicurioParsedSchema(schema);
     } else if (Objects.nonNull(registryName) && registryName.equalsIgnoreCase(SchemaRegistryEnum.CONFLUENT.toString())) {
```

Comment:
We want to avoid have to add new if-else statement if new registry vendors are included. We should use Factory Method